File: apps/web/lib/types/suppliers.ts
export interface SupplierContact {
  id: number
  full_name: string
  role: string
  email: string
  phone: string
  is_primary: boolean
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface Supplier {
  id: number
  display_name: string
  name: string // Phase 1 alias for display_name — remove in Phase 3
  code: string
  legal_name: string
  email: string
  phone: string
  payment_terms_days: number | null
  default_lead_time_days: number | null
  currency: string
  tax_id: string
  address_line1: string
  address_line2: string
  city: string
  state: string
  postal_code: string
  country: string
  notes: string
  is_active: boolean
  contacts: SupplierContact[]
  created_at: string
  updated_at: string
}

File: apps/api/exports/rows/suppliers.py
HEADERS = [
    "code",
    "display_name",
    "legal_name",
    "email",
    "phone",
    "payment_terms_days",
    "default_lead_time_days",
    "currency",
    "country",
    "is_active",
]


def to_row(supplier):
    return [
        supplier.code,
        supplier.display_name,
        supplier.legal_name,
        supplier.email,
        supplier.phone,
        supplier.payment_terms_days if supplier.payment_terms_days is not None else "",
        supplier.default_lead_time_days if supplier.default_lead_time_days is not None else "",
        supplier.currency,
        supplier.country,
        supplier.is_active,
    ]
